### DISCUSSION TITLE
Incremental deployment uses only first file found under factory folder.

### DISCUSSION THREAD

**Discussion by MartaUch** [Azure-Player/azure.datafactory.tools#347] [Q&A] [answered]
Hello,

I'm still investigating how to enable deployment of global parameters for three different environment and usage of incremental deployment feature. In my case I would have three files under /factory folder:
- deep-dev-adf-we
- deep-int-adf-we
- deep-prd-adf-we

My expectation is that during incremental deployment global parameter "adftools_deployment_state" with hashes will be created in each ADF instance after first initial run on each environment.

I tested the approach on POC environment and I'm running into an issue.
I have currently two files under /factory:
- deep-dev-adf-we.json
- deep-poc-dev-adf-we.json

Here are the settings:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/28823253/4f8f7419-67a5-404c-9e49-3acbca761833)
Two files are recognized:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/28823253/76311847-f333-49d3-a32b-3b6adb16324b)
Global parameters are deployed from:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/28823253/1ffa3f73-b0bc-4c54-a398-f844d341c3e3)
Later the step "STEP: Updating (incremental) deployment state..." starts and it takes into account FIRST json file found under the /factory folder:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/28823253/4a23eeb2-3578-4c5e-b122-d60c50121591)
As a result, "adftools_deployment_state" is not deployed to deep-poc-dev-adf-we instance.

I was trying to find this part in the code and what I found so far is that we take always first object from factories:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/28823253/e39b4570-24c1-4a59-855e-07315d54f73a)

**QUESTION**

- Is it possible to change the approach and not assume that there is always one Factory file or it would require a big refactoring of the code?

Thank you in advance!
Marta

**Comment by NowinskiK** [comment, marked as answer]
**Q:** Is it possible to change the approach and not assume that there is always one Factory file or it would require a big refactoring of the code?
**A:** Technically it's possible, but unfortunately there is no plans to go towards this direction. The assumption is that there is ONLY ONE file in factory. It's your risk if you have more. 
**Workaround:** just delete any other files you don't need and run deployment within the tool.

**Reply by MartaUch** [reply]
Alright, I understand, normally I think it's standard that we have one file there, so it's understandable. I was curious if it would be possible to change the approach :)
Thank you for the workaround, I will think about it in addition. Currently I found another approach with usage of one file, so hope it will solve the problem.
